google/cloud/bigquery/_helpers.py

```diff
@@ -172,3 +172,55 @@ def _validate(self, value):
         """
         if value not in self.ALLOWED:
             raise ValueError('Pass one of: %s' ', '.join(self.ALLOWED))
+
+
+class UDFResource(object):
```

Comment:
Was this meant to be in this change? Seems fine to me to move the UDFResource class, but does this refactor belong in the PR to add QueryJob.results?